test/e2e/network_test.go

```diff
@@ -154,6 +155,45 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+// testPodDNSResolution test the DNS resolution in a Windows pod
+func testPodDNSResolution(t *testing.T) {
+	// create context
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+	// the following DNS resolution tests use curl.exe because nslookup tool is not present
+	// in the selected container image for the e2e tests. Ideally, we would use the native
+	// PowerShell cmdlet Resolve-DnsName, but it is not present either.
+	// TODO: Use a compatible container image for the e2e test suite that includes the
+	//  PowerShell cmdlet Resolve-DnsName.
+	t.Run("Valid URL", testCtx.testPodDNSResolutionWithValidUrl)
+	t.Run("Invalid URL", testCtx.testPodDNSResolutionWithInvalidUrl)
+}
+
+// testPodDNSResolutionWithValidUrl verifies that a curl query succeed using a valid URL
+// with resolvable DNS record
+func (tc *testContext) testPodDNSResolutionWithValidUrl(t *testing.T) {
+	winJob, err := tc.createWindowsServerJob("win-dns-tester-valid-url",
+		// curl'ing with --head flag to fetch only the headers as a lightweight alternative
+		"curl.exe --head -k https://www.openshift.com",
+		nil)
+	require.NoError(t, err, "could not create Windows tester job")
+	defer tc.deleteJob(winJob.Name)
+	err = tc.waitUntilJobSucceeds(winJob.Name)
+	assert.NoError(t, err, "Windows tester job failed")
+}
+
+// testPodDNSResolutionWithInvalidUrl verifies that a curl query fail using an invalid URL
+// with unresolvable DNS record
+func (tc *testContext) testPodDNSResolutionWithInvalidUrl(t *testing.T) {
```

Comment:
I feel we should not  have same implementation and two different functions to test valid/invalid case.
Instead pass valid param and expect success and invalid param and expect failure . Like you have done it in func TestGetDNS(). WDYT?

Reply:
@selansen  Thanks for your feedback. Going too deep in generalization may jeopardize the readability of the e2e test. [Before](https://github.com/openshift/windows-machine-config-operator/pull/777#discussion_r749343575)  proposed an approach combining common implementation, but decided to go with this route for simplicity.

Reply:
This is test case so I am good with it previous comment.